src/engraving/infrastructure/io/compat/qfileinfoprovider.h

```diff
@@ -0,0 +1,49 @@
+/*
+ * SPDX-License-Identifier: GPL-3.0-only
+ * MuseScore-CLA-applies
+ *
+ * MuseScore
+ * Music Composition & Notation
+ *
+ * Copyright (C) 2021 MuseScore BVBA and others
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+#ifndef MU_ENGRAVING_QFILEINFOPROVIDER_H
+#define MU_ENGRAVING_QFILEINFOPROVIDER_H
+
+#include "../ifileinfoprovider.h"
+
+#include <QFileInfo>
+
+namespace mu::engraving {
+class QFileInfoProvider : public IFileInfoProvider
+{
+public:
+    explicit QFileInfoProvider(const QFileInfo& fileInfo);
+    explicit QFileInfoProvider(const QString& filePath);
+
+    QString absoluteDirPath() const override;
```

Comment:
A lot of these methods are already implemented in path.h. So we should use path.h instead of reimplementing these methods here